vulnfeeds/cmd/cvelist2osv/main.go

```diff
@@ -0,0 +1,484 @@
+		r := ""
+		if repos, ok := reposForCVE[CVE]; ok {
+			r = strings.Join(repos, " ")
+		}
+		w.Write([]string{string(CVE), outcome.String(), r})
+	}
+	w.Flush()
+
+	if err = w.Error(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func identifyPossibleURLs(cve cves.CVE5) []cves.Reference {
+
+	refs := cve.Containers.CNA.References
+	// check if there are more references in the ADP
+	for _, adp := range cve.Containers.ADP {
+		if adp.References != nil {
+			refs = append(refs, adp.References...)
+		}
+	}
+	for _, affected := range cve.Containers.CNA.Affected {
+		if affected.CollectionUrl != "" {
+			refs = append(refs, cves.Reference{Url: affected.CollectionUrl})
+		}
+		if affected.Repo != "" {
+			refs = append(refs, cves.Reference{Url: affected.Repo})
+		}
+	}
+
+	slices.SortStableFunc(refs, func(a, b cves.Reference) int {
+		return strings.Compare(a.Url, b.Url)
+	})
+	refs = slices.CompactFunc(refs, func(a, b cves.Reference) bool {
+		return a.Url == b.Url
+	})
+	return refs
+}
+
+func main() {
+	flag.Parse()
+	if !slices.Contains([]string{"OSV", "PackageInfo"}, *outFormat) {
+		fmt.Fprintf(os.Stderr, "Unsupported output format: %s\n", *outFormat)
+		os.Exit(1)
+	}
+
+	Metrics.Outcomes = make(map[cves.CVEID]ConversionOutcome)
+
+	var logCleanup func()
+	Logger, logCleanup = utility.CreateLoggerWrapper("cvelist-osv")
+	defer logCleanup()
+
+	data, err := os.ReadFile(*jsonPath)
+	if err != nil {
+		Logger.Fatalf("Failed to open file: %v", err) // double check this is best practice output
+	}
+
+	var cve cves.CVE5
+	err = json.Unmarshal(data, &cve)
+	if err != nil {
+		Logger.Fatalf("Failed to parse CVEList CVE JSON: %v", err)
+	}
+
+	// VPRepoCache := make(VendorProductToRepoMap)
+	ReposForCVE := make(map[cves.CVEID][]string)
+	refs := identifyPossibleURLs(cve)
+
+	CVEID := cve.Metadata.CVEID
+
+	// Not rejecting cause we might be able to leverage other datasources
+
+	if len(refs) > 0 {
+		repos := cves.ReposFromReferencesCVEList(string(CVEID), nil, nil, refs, RefTagDenyList, Logger)
+		if len(repos) == 0 {
+			Logger.Warnf("[%s]: Failed to derive any repos", CVEID)
+		}
+		Logger.Infof("[%s]: Derived %q for CVE", CVEID, repos)
+		ReposForCVE[CVEID] = repos
+	}
+
+	Logger.Infof("[%s]: Repos: %#v", CVEID, ReposForCVE[CVEID])
+
+	switch *outFormat {
+	case "OSV":
+		err = CVEToOSV(cve, ReposForCVE[CVEID], RepoTagsCache, *outDir)
+
+		// TODO: Implement CVEToPackageInfo for CVEList CVE
+		// case "PackageInfo":
+		// 	err = CVEToPackageInfo(cve.CVE, ReposForCVE[CVEID], RepoTagsCache, *outDir)
+		// }
+		// Parse this error to determine which failure mode it was
+		if err != nil {
+			Logger.Warnf("[%s]: Failed to generate an OSV record: %+v", CVEID, err)
+			if errors.Is(err, ErrNoRanges) {
+				Metrics.Outcomes[CVEID] = NoRanges
+
+			}
+			if errors.Is(err, ErrUnresolvedFix) {
+				Metrics.Outcomes[CVEID] = FixUnresolvable
+
+			}
+			Metrics.Outcomes[CVEID] = ConversionUnknown
```

Comment:
this will override no matter what error we found. Do you want do:
```
if errors.Is(err, ErrNoRanges) {
				Metrics.Outcomes[CVEID] = NoRanges
			} else if errors.Is(err, ErrUnresolvedFix) {
				Metrics.Outcomes[CVEID] = FixUnresolvable
			} else {
                                 Metrics.Outcomes[CVEID] = ConversionUnknown
                         }
```
instead?